### PR TITLE
PLANNER-2327 Use fast-jar docker file only

### PR DESCRIPTION
**Pull request by rsynek**
As we stay on Quarkus 1.11.x for now, we can at least prepare for the future upgrade to 1.12.0 and switch to the fast-jar packaging now.

Apart from switching to the fast-jar by default, this PR also adds these missing dockerfiles:
- vaccination scheduling - jvm, native
- factorio-layout - only jvm, as the native runner is broken https://issues.redhat.com/browse/PLANNER-2339

### JIRA
https://issues.redhat.com/browse/PLANNER-2327

### CODE REVIEW

File: quarkus-school-timetabling/src/main/docker/Dockerfile.jvm [hunk 1]
```diff
@@ -13,19 +13,44 @@
 #
 # docker run -i --rm -p 8080:8080 optaplanner/school-timetabling-jvm
 #
+# If you want to include the debug port into your docker image
```
Comment (rsynek) [inline quarkus-school-timetabling/src/main/docker/Dockerfile.jvm:16]: The dockerfile has been aligned with the latest template.

File: quarkus-maintenance-scheduling/src/main/docker/Dockerfile.jvm [hunk 2]
```diff
@@ -13,20 +13,44 @@
 #
 # docker run -i --rm -p 8080:8080 optaplanner/maintenance-scheduling-jvm
 #
+# If you want to include the debug port into your docker image
```
Comment (rsynek) [inline quarkus-maintenance-scheduling/src/main/docker/Dockerfile.jvm:16]: The dockerfile has been aligned with the latest template.

File: quarkus-factorio-layout/src/main/docker/Dockerfile.jvm [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
 #
-# Before building the container image run:
+# Before building the docker image run:
 #
-# mvn package -Dquarkus.package.type=fast-jar 
+# mvn package
 #
 # Then, build the image with:
 #
-# docker build -f src/main/docker/Dockerfile.fast-jar -t quarkus/kotlin-quarkus-school-timetabling-fast-jar .
```
Comment (rsynek) [inline quarkus-factorio-layout/src/main/docker/Dockerfile.jvm:10]: Github a bit misinterprets what happened: the kotlin-quarkus-school-timetabling/Dockerfile.fast-jvm has been removed and a new Dockerfile for the factorio-layout has been created.

File: build/quickstarts-showcase/src/main/java/org/optaplanner/quickstarts/all/rest/QuickstartLauncherResource.java [hunk 4]
```diff
@@ -147,6 +148,17 @@ public void launchQuickstart(@PathParam("quickstartId") String quickstartId) {
         quickstartMeta.getPorts().add(port);
     }
 
+    private File getQuickstartRunnerJar(String quickstartId) {
+        File quickstartRunnerJar = FileSystems.getDefault().getPath(baseDirectory.getAbsolutePath(), quickstartId,
+                "quarkus-app", "quarkus-run.jar").toFile();
+        if (!quickstartRunnerJar.exists()) {
+            throw new IllegalStateException(
+                    "The quickstart (" + quickstartId + ") runner jar does not exist (" + quickstartRunnerJar.getAbsolutePath()
```
Comment (triceo) [inline build/quickstarts-showcase/src/main/java/org/optaplanner/quickstarts/all/rest/QuickstartLauncherResource.java]:
```suggestion
                    "The quickstart (" + quickstartId + ") runner JAR file does not exist (" + quickstartRunnerJar.getAbsolutePath()
```

**Review by triceo — APPROVED** [review]
If it works, I don't see why not.

**Pull request merged**